src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -416,6 +416,31 @@ public static IntPtr AllocateTypeAssociatedMemory(Type type, int size)
                 }
             }
         }
+
+        public static unsafe object? Box(ref byte target, RuntimeTypeHandle type)
+        {
+            if (type.IsNullHandle())
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.type);
+
+            RuntimeType rtType = (RuntimeType)Type.GetTypeFromHandle(type)!;
+
+            if (rtType.IsByRefLike)
+                throw new NotSupportedException(SR.NotSupported_ByRefLike);
+
+            if (rtType.IsPointer || rtType.IsFunctionPointer)
```

Comment:
CoreCLR has RuntimeTypeHandles for generic parameters too. They should be checked here as well (we should reject all types where `TypeHandle.IsTypeDesc` returns true).
```
Type t = typeof(List<>).GetGenericArguments()[0];
RuntimeTypeHandle rth = Type.GetTypeHandle(t);
Console.WriteLine(rth.Value);
```